Applications/spire/include/spire/book_view/labeled_data_widget.hpp

```diff
@@ -0,0 +1,37 @@
+#ifndef SPIRE_LABELED_DATA_WIDGET_HPP
+#define SPIRE_LABELED_DATA_WIDGET_HPP
+#include <QWidget>
+#include <QLabel>
+
+namespace spire {
+
+  //! Represents a label with a label string and data string.
+  class labeled_data_widget : public QWidget {
+    public:
+
+      //! Constructs a labeled_data_widget with label text and data text.
+      /*
+        \param label_text The label text string.
+        \param data_text The data text value.
+        \param parent The parent to this widget.
+      */
+      labeled_data_widget(const QString& label_text, const QString& data_text,
+        QWidget* parent = nullptr);
+
+      //! Sets the label text.
+      void set_label_text(const QString& text);
+
+      //! Sets the data text.
+      void set_data_text(const QString& text);
+
+      //! Sets the label text and data text.
```

Comment:
Remove this setter. Generally it's not a good idea to provide setters for multiple fields unless there is a requirement that the fields update atomically.

Reply:
Do you mean that I should combine them into one?

Reply:
I meant that each field should have its own setter. There shouldn't be one setter that updates two different fields.

Reply:
Whoops, I had those changes in a local commit that I hadn't pushed.